sentry/src/main/java/io/sentry/Attachment.java

```diff
@@ -0,0 +1,118 @@
+package io.sentry;
+
+import java.io.File;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/** You can use an attachment to store additional files alongside an event or transaction. */
+@ApiStatus.Experimental
+public final class Attachment implements Cloneable {
+
+  private @Nullable byte[] bytes;
+  private @Nullable String path;
+  private final @NotNull String filename;
+  private @NotNull String contentType;
+
+  /**
+   * We could use Files.probeContentType(path) to determine the content type of the filename. This
+   * needs a path, but file.toPath or Paths.get only work on above Android API level 26, see
+   * https://developer.android.com/reference/java/nio/file/Paths. There are also ways via
+   * URLConnection, but we don't want to use this in constructors. Therefore we use the default
+   * content type of Sentry.
+   */
+  private static final String DEFAULT_CONTENT_TYPE = "application/octet-stream";
+
+  /**
+   * Initializes an Attachment with bytes.
+   *
+   * <p>The file located at the path is read lazily when the SDK captures an event or transaction
+   * not when the attachment is initialized.
+   *
+   * @param bytes The bytes of file.
+   * @param filename The name of the attachment to display in Sentry.
+   */
+  public Attachment(final @NotNull byte[] bytes, final @NotNull String filename) {
+    this.bytes = bytes;
+    this.filename = filename;
+    this.contentType = DEFAULT_CONTENT_TYPE;
+  }
+
+  /**
+   * Initializes an Attachment with a path. The filename of the file located at the path is used.
+   *
+   * @param path The path of the file to upload as an attachment.
+   */
```

Comment:
L: maybe include that we also require an `absolute` path? or rename the String param to make it obvious

Reply:
I renamed `path` to `pathname` and also updated the javadoc. This is more aligned with [`new File(String pathname)`](https://docs.oracle.com/javase/8/docs/api/java/io/File.html#File-java.lang.String-). It doesn't have to be an absolute path. Is it clearer now?

Reply:
Files on Android are a bit messy, for example, a `pathname` from a taken picture simply doesn't work well if you don't resolve full address with the `MediaStore` intent classes (Uri etc), also it changes by OS versions.

An important note is that this should be run on a Java App as well which would be good enough already.

If people are familiar with the File APIs, it's good already, I was stressing only the docs.